cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/distributed/DistributedSparkProgramRunner.java

```diff
@@ -115,6 +116,7 @@ private static YarnConfiguration createConfiguration(YarnConfiguration hConf, CC
                                                        TokenSecureStoreRenewer secureStoreRenewer) {
     YarnConfiguration configuration = new YarnConfiguration(hConf);
     configuration.setBoolean(SparkRuntimeContextConfig.HCONF_ATTR_CLUSTER_MODE, true);
+    configuration.set("hive.metastore.token.signature", HiveAuthFactory.HS2_CLIENT_TOKEN);
```

Comment:
This was added as a support for spark2 on cdh here https://github.com/caskdata/cdap/commit/4b8b2eac210a65b5fdd36094cac7e64e3e48ac99#diff-9e51b61e5d3c6758ba9a00dcefd7a497R128